File: alphatimer-protocol-api/src/main/java/org/lisasp/alphatimer/api/protocol/events/dropped/DataHandlingMessage2DroppedEvent.java
package org.lisasp.alphatimer.api.protocol.events.dropped;

import lombok.RequiredArgsConstructor;
import lombok.Value;
import org.lisasp.alphatimer.api.protocol.data.Bytes;

import java.time.LocalDateTime;

@Value
@RequiredArgsConstructor
public class DataHandlingMessage2DroppedEvent implements InputDataDroppedEvent {
    private final LocalDateTime timestamp;
    private final String competition;
    private final String message;
    private final Bytes data;

    public DataHandlingMessage2DroppedEvent(LocalDateTime timestamp, String competition, String message, byte[] data) {
        this(timestamp, competition, message, new Bytes(data));
    }
}

File: protocol-api/src/main/java/de/dennisfabri/alphatimer/api/protocol/events/dropped/DataHandlingMessage2DroppedEvent.kt
package de.dennisfabri.alphatimer.api.protocol.events.dropped

import de.dennisfabri.alphatimer.api.protocol.data.Bytes

data class DataHandlingMessage2DroppedEvent(private val message: String, private val data: Bytes) : InputDataDroppedEvent {
    constructor(message: String, data: ByteArray) : this(message, Bytes(data))
}
